builder/azure/common/pointer.go

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+//go:generate packer-sdc struct-markdown
+
+package common
+
+// StringPtr returns a pointer to the passed string.
+func StringPtr(s string) *string {
```

Comment:
I'm surprised we have to make our own, isn't there something like this in the Azure SDK already?

Reply:
Might be a good idea to push it to the Hashicorp Azure SDK if not, especially if there's a bunch of string pointers in the API calls.

Reply:
This was in the old SDK as an annoying helper tbh, and I didn't want to have to have a bunch of duplicated code.  The new SDK does not have this and tbh I wasn't sure how it'd be received as a PR so I felt this was simpler